jsurfer-core/src/main/java/org/jsfr/json/compiler/JsonPathCompiler.java

```diff
@@ -314,6 +319,37 @@ public Void visitFilterMatchRegex(JsonPathParser.FilterMatchRegexContext ctx) {
         return rst;
     }
 
+    @Override
+    public Void visitFilterEqualType(JsonPathParser.FilterEqualTypeContext ctx) {
+        filterPathBuilder = createFilterPathBuilder();
+        Void rst = super.visitFilterEqualType(ctx);
+        ItemMethod itemMethod = getItemMethod(ctx.itemMethod());
+        if (itemMethod == ItemMethod.TYPE) {
+            Type type = getType(ctx.QUOTED_STRING());
+            filterBuilder.append(new EqualityTypePredicate(filterPathBuilder.build(), type));
+        } else {
+            throw new InputMismatchException("Unsupported item method: " + itemMethod);
+        }
+        filterPathBuilder = null;
+        return rst;
+    }
+
+    @Override
+    public Void visitFilterNEqualType(JsonPathParser.FilterNEqualTypeContext ctx) {
```

Comment:
We can unify `visitFilterNEqualType` and `visitFilterEqualType` with one boolean parameter.

Reply:
I have tried it but it does not worth it. Too many different lines.